src/ORM/Association/BelongsToMany.php

```diff
@@ -1278,12 +1256,26 @@ protected function _buildQuery($options)
 
         // Ensure that association conditions are applied
         // and that the required keys are in the selected columns.
+
+        $tempName = $this->_name . '_CJoin';
+        $schema = $assoc->schema();
```

Comment:
Should we make a local reference to the `target`? There are a few uses of it now.